src/flint/types/fq_default_poly.pyx

```diff
@@ -0,0 +1,1655 @@
+        # set a constant polynomial
+        v = self.field.any_as_fq_default(obj)
+        if v is NotImplemented:
+            return NotImplemented
+
+        fq_default_poly_set_fq_default(poly, (<fq_default>v).val, self.field.val)
+
+    cdef any_as_fq_default_poly(self, obj):
+        # If fq_default_poly, return if fields match
+        if typecheck(obj, fq_default_poly):
+            if self != (<fq_default_poly>obj).ctx:
+                raise ValueError("fields must match")
+            return obj
+
+        # For all other types attempt to coerce to a fq_default_poly
+        cdef fq_default_poly res
+        res = self.new_ctype_poly()
+        check = self.set_any_as_fq_default_poly(res.val, obj)
+        if check is NotImplemented:
+            return NotImplemented
+        return res
+
+    def base_field(self):
+        """
+        Return the base field of the polynomial ring
+
+            >>> R = fq_default_poly_ctx(65537, 3)
+            >>> R.base_field()
+            fq_default_ctx(65537, 3, 'z', x^3 + 3*x^2 + 30077, 'FQ_NMOD')
+
+        """
+        return self.field
+
+    def characteristic(self):
+        """
+        Return the characteristic of the field from the context
+        as an ``fmpz`` type
+
+            >>> R = fq_default_poly_ctx(65537, 3)
+            >>> R.characteristic()
+            65537
+
+        """
+        return self.field.characteristic()
+
+    prime = characteristic
+
+    def zero(self):
+        """
+        Return the zero element of this polynomial ring
+
+            >>> R = fq_default_poly_ctx(163)
+            >>> R.zero()
+            0
+        """
+        cdef fq_default_poly res
+        res = self.new_ctype_poly()
+        fq_default_poly_zero(res.val, self.field.val)
+        return res
+
+    def one(self):
+        """
+        Return the one element of this polynomial ring
+
+            >>> R = fq_default_poly_ctx(163)
+            >>> R.one()
+            1
+        """
+        cdef fq_default_poly res
+        res = self.new_ctype_poly()
+        fq_default_poly_one(res.val, self.field.val)
+        return res
+
+    def gen(self):
+        """
+        Return the generator of the polynomial: `x`
+
+            >>> R = fq_default_poly_ctx(163)
+            >>> R.gen()
+            x
+        """
+        cdef fq_default_poly res
+        res = self.new_ctype_poly()
+        fq_default_poly_gen(res.val, self.field.val)
+        return res
+
+    def random_element(self, degree=3, not_zero=False, monic=False, irreducible=False):
+        """
+        Return a random element of degree ``degree``. If ``not_zero``
+        is ``True``, ensures the output is not zero, if ``monic`` is
+        ``True``, ensures the output is monic. If ``irreducible`` is
+        ``True``, ensures that the output is irreducible.
+
+        TODO: there is currently no fq_defualt_poly method for testing that
+        a polynomial is irreducible?!
```

Comment:
We can add one here.

Apart from shortcutting some simple cases the general way to check is just to factorise the polynomial.

Reply:
Oh this is outdated there is one it was just in the poly_factor header file rather than the poly header so I missed it when I was writing this.